app/models/course.rb

```diff
@@ -113,6 +113,23 @@ def root_folder?
     end
   end
 
+  def self_directed_learning_max_time
+    settings(:course).self_directed_learning_max_time
+  end
+
+  def self_directed_learning_max_time=(time)
```

Comment:
@weiqingtoh @fonglh  this is the max `duration` of the mode.  so `acutal_start_at = start_at - duration`. but `earlier_start_at` or `advanced_start_at` are more like a time point ( other than a duration)